### ISSUE TITLE
Unable to initialize

### ISSUE THREAD

**Issue by lekoook** [sparkfun/SparkFun_MAG3110_Breakout_Board_Arduino_Library#4]
The initialize() function always return false. .dataReady() always return false as well, thus no readings can be made. If I try to read the values directly, the arduino seems to just hang.

Does anyone face the same issue?

**Comment by bboyho** [comment]
Hi,

I was fixing another issue in this library when I noticed your question. I did not have any issue reading the sensor. If you have not tried yet, make sure to [check your solder joints](https://learn.sparkfun.com/tutorials/sparkfun-troubleshooting-tips#bad-solder-joints) and ensure to remove any [flux residue of you are using water soluble flux](https://learn.sparkfun.com/tutorials/sparkfun-troubleshooting-tips#flux-residue) on the hardware side.

It also looks like the sensor was tested on an AVR. I'm not sure what development board you are using. Try moving the following lines out of the &#42;.cpp file and into the example sketches after the `Serial.begin()` line. Based on previous issues that I have seen, it won't work before for certain development boards. 

    Wire.begin(); //setup I2C bus
    Wire.setClock(400000);    // I2C fast mode, 400kHz
